torch/_tensor_str.py

```diff
@@ -239,9 +239,17 @@ def get_summarized_data(self):
 def _str(self):
     prefix = 'tensor('
     indent = len(prefix)
-
     suffixes = []
-    if not torch._C._is_default_type_cuda():
+
+    # Note [Print tensor device]:
+    # A general logic here is we only print device when it doesn't match
+    # the device specified in default tensor type.
+    # Currently torch.set_default_tensor_type() only supports CPU/CUDA.
+    # In other cases, we don't have a way to set them as default yet,
+    # and we should always print out device for them.
+    if self.device.type == 'xla':
+        suffixes.append('device=\'' + str(self.device) + '\'')
+    elif not torch._C._is_default_type_cuda():
```

Comment:
Perhaps we could have a method that queries the default device instead, and prints this block if the tensor device is not that one? That would make it robust to all kinds of choices, and we wouldn't have to revisit this code in the future.